### ISSUE TITLE
Does not work with NPM and Webpack

### ISSUE THREAD

**Issue by cgarvis** [fmoo/react-typeahead#39]
```
ERROR in ./~/react-typeahead/src/typeahead/index.js                                                                                                                                                        │
Module parse failed: /Users/cgarvis/projects/peach.corp/node_modules/react-typeahead/src/typeahead/index.js Line 90: Unexpected token <                                                                    │
You may need an appropriate loader to handle this file type.                                                                                                                                               │
|       ) {                                                                                                                                                                                                │
|       return (                                                                                                                                                                                           │
|         <TypeaheadSelector                                                                                                                                                                               │
|           ref="sel" options={this.state.visible}                                                                                                                                                         │
|           customValue={this.state.entryValue}                                                                                                                                                            │
 @ ./~/react-typeahead/src/react-typeahead.js 1:16-38                                                                                                                                                      │
                                                                                                                                                                                                           │
ERROR in ./~/react-typeahead/src/tokenizer/index.js                                                                                                                                                        │
Module parse failed: /Users/cgarvis/projects/peach.corp/node_modules/react-typeahead/src/tokenizer/index.js Line 55: Unexpected token <                                                                    │
You may need an appropriate loader to handle this file type.                                                                                                                                               │
|     var result = this.state.selected.map(function(selected) {                                                                                                                                            │
|       return (                                                                                                                                                                                           │
|         <Token key={ selected } className={classList}                                                                                                                                                    │
|           onRemove={ this._removeTokenForValue }                                                                                                                                                         │
|           name={ this.props.name }>                                                                                                                                                                      │
 @ ./~/react-typeahead/src/react-typeahead.js 2:16-38
```

**Comment by fmoo** [comment]
@cgarvis - what command are you running to trigger this?  I'm guessing this has something to do with using jsx syntax in a .js file.

**Comment by cgarvis** [comment]
I'm using the `babel-loader` with webpack so I'm using es6 style of requiring modules.

Require in the app:

`import Typeahead from 'react-typeahead';`

My webpack config

```javascript
var webpack = require('webpack');                                                                                                                                                                          │
var HtmlWebpackPlugin = require('html-webpack-plugin')                                                                                                                                                     │
                                                                                                                                                                                                           │
module.exports = {                                                                                                                                                                                         │
  entry: './src/index',                                                                                                                                                                                    │
  output: {                                                                                                                                                                                                │
    path: __dirname + '/dist',                                                                                                                                                                             │
    filename: 'bundle.[hash].js',                                                                                                                                                                          │
    publicPath: '/',                                                                                                                                                                                       │
  },                                                                                                                                                                                                       │
  resolve: {                                                                                                                                                                                               │
    extensions: ['', '.js']                                                                                                                                                                                │
  },                                                                                                                                                                                                       │
  resolveLoader: {                                                                                                                                                                                         │
    modulesDirectories: ['node_modules']                                                                                                                                                                   │
  },                                                                                                                                                                                                       │
  plugins: [                                                                                                                                                                                               │
    new HtmlWebpackPlugin({                                                                                                                                                                                │
      title: 'Peach for Businesses',                                                                                                                                                                       │
      template: 'src/index.html',                                                                                                                                                                          │
    }),                                                                                                                                                                                                    │
    new webpack.DefinePlugin({                                                                                                                                                                             │
      "process.env.NODE_ENV": "'" + process.env.NODE_ENV + "'"                                                                                                                                             │
    })                                                                                                                                                                                                     │
  ],                                                                                                                                                                                                       │
  module: {                                                                                                                                                                                                │
    loaders: [                                                                                                                                                                                             │
      { test: /\.js?$/, loaders: ['babel-loader'], exclude: /node_modules/ }                                                                                                                               │
    ]                                                                                                                                                                                                      │
  }                                                                                                                                                                                                        │
};
```

**Comment by raysuelzer** [comment]
I don't think babel handles jsx transformations?

**Comment by cgarvis** [comment]
It does.  I think the problem is that I don't have it running over the node modules.

**Comment by raysuelzer** [comment]
The regex for you loader appears to be wrong. I tested it and it doesn't
match .jsx files.  Try this

      { test: /\.js.$/, loaders: ['babel-loader'], exclude: /node_modules/ }


Ray Suelzer
Mobile: (312) 869-2665

**Comment by raysuelzer** [comment]
Sorry, that regex will miss .js files.  Either way, the regex for the loader is the root of the problem here.

**Comment by cgarvis** [comment]
The issue is more that when the lib is installed via npm, the code is raw jsx.  What do you think of transforming it to straight js when you publish to npm?  The benefit is that I don't have to run babel over the node module dir.

**Comment by fmoo** [comment]
@cgarvis - definitely think that's the right approach here.  Only two things I'm not sure about:

1. Is it as easy as updating main in the package.js?  Or will that break other things?
2.  I think the compiled version includes dependencies as well.

**Comment by fmoo** [comment]
iiuc, this has been fixed with #49
